src/ClientAppSpec.ts

```diff
@@ -139,6 +139,90 @@ export default describe('ClientApp', () => {
                 });
             });
         });
+        describe('gcHostOriginQueryParam and gcTargetEnvQueryParam config', () => {
+            it('should allow a user to pass valid query param names into the constructor', () => {
+                // External
+                let query = '?gcHostOrigin=https://apps.mypurecloud.com&gcTargetEnv=prod';
+                spyOn(ClientApp, '_getQueryString').and.callFake(() => query);
+                let myClientApp = new ClientApp({
+                    gcHostOriginQueryParam: 'gcHostOrigin',
+                    gcTargetEnvQueryParam: 'gcTargetEnv'
+                });
+                expect(myClientApp.pcEnvironment).toBe('mypurecloud.com');
+                // Localhost
+                query = '?gcHostOrigin=https://localhost:1337&gcTargetEnv=prod';
+                myClientApp = new ClientApp({
+                    gcHostOriginQueryParam: 'gcHostOrigin',
+                    gcTargetEnvQueryParam: 'gcTargetEnv'
+                });
+                expect(myClientApp.pcEnvironment).toBe('localhost');
+            });
+            it('should fail if is not targeting a valid env', () => {
+                // External
+                let query = '?gcHostOrigin=https://invalid.com&gcTargetEnv=prod';
+                spyOn(ClientApp, '_getQueryString').and.callFake(() => query);
+                expect(() => {
+                    new ClientApp({
+                        gcHostOriginQueryParam: 'gcHostOrigin',
+                        gcTargetEnvQueryParam: 'gcTargetEnv'
+                    });
+                }).toThrow();
+                // Localhost
+                query = '?gcHostOrigin=https://apps.mypurecloud.com&gcTargetEnv=invalid';
+                expect(() => {
+                    new ClientApp({
+                        gcHostOriginQueryParam: 'gcHostOrigin',
+                        gcTargetEnvQueryParam: 'gcTargetEnv'
+                    });
+                }).toThrow();
```

Comment:
can we add 1 more case here?  A valid origin and env but mis-matched?

Ex:
 query = '?gcHostOrigin=https://apps.mypurecloud.jp&gcTargetEnv=prod';